bindings/azure/eventhubs/eventhubs_integration_test.go

```diff
@@ -0,0 +1,107 @@
+//go:build integration_test
+// +build integration_test
+
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+package eventhubs
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/kit/logger"
+)
+
+const (
+	// Note: Reuse the environment variables names from the conformance tests where possible to support reuse of setup workflows
+
+	// iotHubConnectionStringEnvKey defines the key containing the integration test connection string
+	// For the default EventHub endpoint for an Azure IoT Hub, it will resemble:
+	// Endpoint=sb://<iotHubGeneratedNamespace>.servicebus.windows.net/;SharedAccessKeyName=service;SharedAccessKey=<key>;EntityPath=<iotHubGeneratedPath>
+	iotHubConnectionStringEnvKey = "AzureIotHubEventHubConnectionString"
+	iotHubConsumerGroupEnvKey    = "AzureIotHubBindingsConsumerGroup"
+	iotHubNameEnvKey             = "AzureIotHubName"
+	storageAccountNameEnvKey     = "AzureBlobStorageAccount"
+	storageAccountKeyEnvKey      = "AzureBlobStorageAccessKey"
+	azureCredentialsEnvKey       = "AZURE_CREDENTIALS"
+
+	testStorageContainerName = "iothub-bindings-integration-test"
+)
+
+func createIotHubBindingsMetadata() bindings.Metadata {
+	metadata := bindings.Metadata{
+		Properties: map[string]string{
+			connectionString:     os.Getenv(iotHubConnectionStringEnvKey),
+			consumerGroup:        os.Getenv(iotHubConsumerGroupEnvKey),
+			storageAccountName:   os.Getenv(storageAccountNameEnvKey),
+			storageAccountKey:    os.Getenv(storageAccountKeyEnvKey),
+			storageContainerName: testStorageContainerName,
+		},
+	}
+
+	return metadata
+}
+
+func testReadIotHubEvents(t *testing.T) {
+	logger := logger.NewLogger("bindings.azure.eventhubs.integration.test")
+	eh := NewAzureEventHubs(logger)
+	err := eh.Init(createIotHubBindingsMetadata())
+	assert.Nil(t, err)
+
+	// Invoke az CLI via bash script to send test IoT device events
+	// Requires the AZURE_CREDENTIALS environment variable to be already set (output of `az ad sp create-for-rbac`)
+	cmd := exec.Command("/bin/bash", "../../../tests/scripts/send-iot-device-events.sh")
```

Comment:
Is there a way to make this work based on the repo "root" instead of this folder?

Reply:
I'm not keen on relative paths here, but I couldn't think of a good, simple way to reference a file elsewhere in the git tree. Alternatives considered:
- shell out and run `git rev-parse --show-toplevel` and then build the the path from the git root. 
- require the caller provide the script path as an env variable
- copy the script to each test folder.

I didn't think it was worth spending too much time on, particularly if this would be rewritten to a different framework (and likely relocated) at that point.